src/parser/expression/abstract_expression.cpp

```diff
@@ -142,4 +161,53 @@ std::shared_ptr<AbstractExpression> DeserializeExpression(const nlohmann::json &
   return expr;
 }
 
+bool AbstractExpression::DeriveSubqueryFlag() {
+  if (expression_type_ == ExpressionType::ROW_SUBQUERY) {
+    has_subquery_ = true;
+  } else {
+    for (auto &child : children_) {
+      if (child->DeriveSubqueryFlag()) {
+        has_subquery_ = true;
+        break;
+      }
+    }
+  }
+  return has_subquery_;
+}
+
+int AbstractExpression::DeriveDepth() {
+  if (depth_ < 0) {
+    for (auto &child : children_) {
+      auto child_depth = child->DeriveDepth();
+      if (child_depth >= 0 && (depth_ == -1 || child_depth < depth_)) depth_ = child_depth;
+    }
+  }
+  return depth_;
+}
+
+void AbstractExpression::DeduceExpressionName() {
+  // If alias exists, it will be used in TrafficCop
+  if (!alias_.empty()) return;
+
+  bool first = true;
+
+  if (expression_type_ == ExpressionType::FUNCTION) {
```

Comment:
Why is the logic for `ExpressionType::FUNCTION` here? Shouldn't it be in `FunctionExpression` in an overrided method similar to how you do it for `ConstantValueExpression`